cli/src/main/java/com/devonfw/tools/ide/tool/mvn/Mvn.java

```diff
@@ -154,12 +153,11 @@ private String getEncryptedPassword(String variable) {
 
     String input = this.context.askForInput("Please enter secret value for variable " + variable + ":");
 
-    ProcessContext pc = this.context.newProcess().executable("mvn");
-    pc.addArgs("--encrypt-password", input);
-    pc.addArg(getSettingsSecurityProperty());
-    ProcessResult result = pc.run(ProcessMode.DEFAULT_CAPTURE);
+    Mvn mvn = new Mvn(context);
+    ProcessResult result = mvn.runTool(ProcessMode.DEFAULT_CAPTURE, ProcessErrorHandling.LOG_WARNING, this.context.newProcess(), "--encrypt-password", input,
+        getSettingsSecurityProperty());
 
-    String encryptedPassword = result.getOut().get(0);
+    String encryptedPassword = result.getOut().getFirst();
```

Comment:
Maybe we can even extract this logic to a private method. Except for the arguments this is exactly the same as for `--encrypt-master-password`.

Reply:
Refactored into `retrievePassword` method now.

Reply:
I'm not sure about using log level `WARNING` for the logger though.